fuzzers/forkserver_libafl_cc/src/main.rs

```diff
@@ -179,7 +179,7 @@ pub fn main() {
     if let Some(dynamic_map_size) = executor.coverage_map_size() {
         executor
             .observers_mut()
-            .match_name_mut::<HitcountsMapObserver<StdMapObserver<'_, u8, false>>>("shared_mem")
+            .match_name_mut::<ExplicitTracking<HitcountsMapObserver<StdMapObserver<'_, u8, false>>, true, false>>("shared_mem")
```

Comment:
Did we make this harder to use instead of easier?

Reply:
yes

Reply:
it's ofcourse more complicated if we add more trait generics to everything when we already had enough of them. previously it was just two additional true false flag to the feedback

Reply:
imo i don't think #1886 is worth the complexity if the purpose of it is just to elimitate that too true, false for mapfeedback

Reply:
Literally no. There is very little reason to ever explicitly refer to the observer type.

The change makes it categorically easier to use, unless you are doing something exceptionally atypical.

Reply:
Also, this was a side effect that I mentioned in the original pull request (i.e., wrapping types). I don't understand why this is now an issue if it was okay when I submitted the pull request.

Reply:
Sorry, I'm just frustrated. Lots of my changes have been having this kind of issue recently.

If we should revert this, then okay, but we should communicate better about side effects like this going forward. I still think this is a relatively elegant solution and that in this case we just need a better way to have interaction between the executor and the observer here that removes the need for explicit observer access. It really feels like explicit observer access is indicative of a problem of how this was implemented, since we have these relatively heavy types.

Reply:
The `.match_name_mut::<...>` that this PR attempted to fix (just saw that the fix is broken :pensive:)

Reply:
in this code we are trying to find the right observer associated to the executor that has the right name and observer. but i think we cannot avoid specifiying its full type name ?

Reply:
See: https://github.com/AFLplusplus/LibAFL/issues/2064

Reply:
what will the code look like after #2064 is resolved?
like this?
```
executor
            .observers_mut()
            .match_name_by_ref("edges", &edges_observer)
```

Reply:
ok looks good

Reply:
Note that obs_ref is a [ZST](https://doc.rust-lang.org/nomicon/exotic-sizes.html) and won't affect runtime performance. This all gets elided.

Reply:
> ```rust
> let obs_ref = edges_observer.make_reference();
> ...
> executor
>   .observers_mut()
>   .match_name_by_ref("edges", obs_ref)
>   .as_mut()
>   .truncate(...);
> ```

If this works it's a great solution! :)

Reply:
> Also, this was a side effect that I mentioned in the original pull request (i.e., wrapping types). I don't understand why this is now an issue if it was okay when I submitted the pull request.

In our defense, this part was missing in the pull request :P